### PR TITLE
Add vmtouch

### PR DESCRIPTION
**Pull request by ressy**
This adds [vmtouch](https://github.com/hoytech/vmtouch), a file system cache utility.

### CODE REVIEW

**Comment by ressy** [comment]
Whoops, just saw https://github.com/hoytech/vmtouch/pull/71 to fix the `make install` failure on Mac OS, which is too new for the latest release.  That explains [the osx_64 failure](https://github.com/conda-forge/staged-recipes/pull/7843/checks?check_run_id=67670373) here.

I could skip the second call to make and just issue the mkdir/install commands as part of the recipe here, so the recipe is still using the unmodified release tarball.  Does that make sense?

**Comment by ressy** [comment]
There we go!  Checks pass on both Linux and Mac OS.  My lingering questions are:

 * Does that substitute for `make install` in the recipe look OK?
 * vmtouch provides a man page, so I install it, but I haven't provided activate/deactivate scripts to munge MANPATH and make it available automatically.  Is that best left to the end user managing the environment, or should the package provide it?

File: recipes/vmtouch/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Copyright (c) 2009-2017 Doug Hoyte and contributors. All rights reserved.
```
Comment (chrisburr) [inline recipes/vmtouch/LICENSE.txt]: This file isn't needed as it is included in the source tarball

File: recipes/vmtouch/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  commands:
+    # Attempt to run vmtouch on itself.  Assume success so long as we have a
+    # zero exit code.
+    - vmtouch -t $PREFIX/bin/vmtouch
+
+about:
+  home: https://hoytech.com/vmtouch/
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.txt
```
Comment (chrisburr) [inline recipes/vmtouch/meta.yaml]: See above:
```suggestion
  license_file: LICENSE
```

File: recipes/vmtouch/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+build:
+  skip: true  # [win]
+  number: 0
+  script:
+    - make
+    # "make install" is broken on Mac OS in version 1.3.1, but should be fixed
+    # in the next release: https://github.com/hoytech/vmtouch/pull/71
+    # Here we'll just include the fix manually.
+    - mkdir -p $PREFIX/bin $PREFIX/share/man/man8
```
Comment (chrisburr) [inline recipes/vmtouch/meta.yaml]: There is a discussion about conda provided man pages [here](https://github.com/conda/conda/issues/845). I think you're doing the "right" thing already.

File: recipes/vmtouch/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+source:
+  url: https://github.com/hoytech/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: d57b7b3ae1146c4516429ab7d6db6f2122401db814ddd9cdaad10980e9c8428c
+
+build:
+  skip: true  # [win]
+  number: 0
+  script:
+    - make
```
Comment (chrisburr) [inline recipes/vmtouch/meta.yaml]: You could use selectors to conditionally run `make install` here:
```suggestion
    - make
    - make install  # [not osx]
```
The macOS specific lines below can then be appended with `  # [osx]`.

**Comment by ressy** [comment]
Thanks @chrisburr!  I was interrupted partway through looking over your changes so thanks for fixing the remainder.

**Pull request merged**